conn.go

```diff
@@ -1194,12 +1194,16 @@ func (c *Conn) executeQuery(ctx context.Context, qry *Query) *Iter {
 		}
 
 		if x.meta.morePages() && !qry.disableAutoPage {
+			newQry := new(Query)
+			*newQry = *qry
+			newQry.pageState = copyBytes(x.meta.pagingState)
+			newQry.metrics = &queryMetrics{m: make(map[string]*hostMetrics)}
```

Comment:
nitpick: wdyt about preallocating map with the same size as `len(qry.metrics.m)`? I'm not sure how metrics are handled, so maybe it doesn't help anything.

Reply:
The map stores metrics per connect address, so the size will be 1 in most cases and will vary based on hosts that will be used for retries. I guess we can leave the capacity at default. If we carried the capacity from previous query, a single query of iterator page that needed to be retried on many hosts (thus its map will be larger) would make us allocate larger maps until the iterator is exhausted even if the subsequent queries don't need to be retried.